results/docs/Control_benchmarks_analysis.md

```diff
@@ -0,0 +1,34 @@
+# Control Benchmarks 
+
+The suite of control conditional benchmarks which is a part of the [university of wisconsin micro-benchmarks suite](https://github.com/VerticalResearchGroup/microbench) can be used to try and understand the difference in performance between the intel skylake processor and the branch prediction unit in gem5.
+
+Control Conditional Benchmarks:
+
+- CCa : completely baised branches
+
+- CCe : easy to predict -- branch pattern (10101010...)
+
+- CCm : heavly baised branches
+
+- CCh_st : impossible to predict with a store operation
+
+
+**Understanding the role of Branch Miss-Predictions**
+
+Inorder to understand the role of branch predictors, we take a  closer look into it by ploting the conditional branch misprediction rate while using LTAGE branchmispredictor (which is the best branchpredictor)
+
+The misprediction rate is almost same as the intel pcm for both UnCalibrated and Calibrated CPU. 
+
+While increasing the iteration to '500K', the misprediction comes down for gem5 while using LTAGE because LtAGE has a bigger history table, therefore gets better with increase in iterations.
+
+<img src="../images/Branchpred_misprediction_100K.png" width="500" height="500">
+
+<img src="../images/Branchpred_misprediction_500K.png" width="500" height="500">
+
+## Conclusion:
+
+
+<img src="../images/Branchpred_ipc_100K.png" width="500" height="500">
+
+As we can see, the IPC for gem5 configured to model skylake architecture with LTAGE branchPredictor for '100K' iteration of the above benchmarks is almost same as the performance of the Intel processor measured using intel pcm for 'Calib CPU'. This proves that after Calibrating the CPU w.r.t execution benchmarks the gem5 performance matches the performance of real hardware system.
```

Comment:
Calib CPU is 7 issue wide whereas as uncalib CPU is 4 issue wide, that's the only reason for the difference. However, BP parameters must be changed in gem5 to study w.rt hardware.

Reply:
but u can see the misprediction graph.. and this conclusion (that the branch predictor settings is ok for now)  was drawn from that as well, not just the IPC. So, I thought, we can add other information and modify the results if we find something new on the go. :)

Reply:
added the extra piece of info u said, we can add more stuff as we perform the other experiments.